delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/routes/CompositeViewsRoutesSpec.scala

```diff
@@ -244,33 +239,68 @@ class CompositeViewsRoutesSpec extends CompositeViewsRoutesFixtures {
       )
 
       forAll(endpoints) { endpoint =>
-        Post(endpoint, esQuery.asJson)(CirceMarshalling.jsonMarshaller, ec) ~> asBob ~> routes ~> check {
+        Post(endpoint, esQuery.asJson)(CirceMarshalling.jsonMarshaller, ec) ~> routes ~> check {
           response.status shouldEqual StatusCodes.OK
           response.asJson shouldEqual esResult
         }
       }
     }
 
     "fail to deprecate a view without permission" in {
-      Delete(s"/v1/views/myorg/myproj/$uuid?rev=3") ~> routes ~> check {
+      Delete(s"/v1/views/myorg/myproj/$uuid?rev=3") ~> asReader ~> routes ~> check {
         response.shouldBeForbidden
       }
     }
 
     "reject a deprecation of a view without rev" in {
-      Delete(s"/v1/views/myorg/myproj/$uuid") ~> asBob ~> routes ~> check {
+      Delete(s"/v1/views/myorg/myproj/$uuid") ~> asWriter ~> routes ~> check {
         response.status shouldEqual StatusCodes.BadRequest
         response.asJson shouldEqual jsonContentOf("routes/errors/missing-query-param.json", "field" -> "rev")
       }
     }
 
     "deprecate a view" in {
-      Delete(s"/v1/views/myorg/myproj/$uuid?rev=3") ~> asBob ~> routes ~> check {
+      Delete(s"/v1/views/myorg/myproj/$uuid?rev=3") ~> asWriter ~> routes ~> check {
         response.status shouldEqual StatusCodes.OK
         response.asJson shouldEqual viewMetadata(4, true)
       }
     }
 
+    "fail to undeprecate a view without permission" in {
+      givenADeprecatedView { view =>
+        Put(s"/v1/views/myorg/myproj/$view/undeprecate?rev=2") ~> asReader ~> routes ~> check {
+          response.shouldBeForbidden
+        }
+      }
+    }
+
+    "reject an undeprecation of a view without rev" in {
+      givenADeprecatedView { view =>
+        Put(s"/v1/views/myorg/myproj/$view/undeprecate") ~> asWriter ~> routes ~> check {
+          response.status shouldEqual StatusCodes.BadRequest
+          response.asJson shouldEqual jsonContentOf("routes/errors/missing-query-param.json", "field" -> "rev")
+        }
+      }
+    }
+
+    "reject an undeprecation of a view that is not deprecated" in {
+      givenAView { view =>
+        Put(s"/v1/views/myorg/myproj/$view/undeprecate?rev=1") ~> asWriter ~> routes ~> check {
+          response.status shouldEqual StatusCodes.BadRequest
+          response.asJson shouldEqual jsonContentOf("routes/errors/view-not-deprecated.json", "id" -> (nxv + view))
```

Comment:
`jsonContentOf("routes/errors/view-not-deprecated.json", "id" -> (nxv + view))` could be `viewNotDeprecatedError(nxv + view)`, and even better: that method could use `json""` rather than loading a file